kbfsblock/id.go

```diff
@@ -91,6 +104,37 @@ func MakeTemporaryID() (ID, error) {
 	return ID{h}, nil
 }
 
+// MakeRandomIDInRange generates a random block ID using a CSPRNG, distributing
+// the random variable over the interval [start, end), where the full range is
+// [0, MaxUint64). This corresponds to a normalized representation of the
+// range [kbfshash.RawDefaultHash{}, kbfshash.MaxDefaultHash).
+func MakeRandomIDInRange(start, end uint64) (ID, error) {
+	if start >= end {
+		return ID{}, errors.New("Invalid range for random ID")
+	}
+	rangeSize := float64(end - start)
+	randBuf := make([]byte, 8)
+	err := kbfscrypto.RandRead(randBuf)
+	if err != nil {
+		return ID{}, err
+	}
+	// Generate a random unsigned int. Endianness doesn't matter here because
+	// the bytes are random.
+	randUint := binary.BigEndian.Uint64(randBuf)
+	randFloat := float64(randUint) / float64(math.MaxUint64)
```

Comment:
Oh I see, from your use of it, it looks like accuracy doesn't matter that much. Perhaps change the API to reflect that, i.e. take in floats between 0 and 1 and mention in the comments that the distribution is only approximately uniform? Would also save conversions from/to uint64